mods/beds/functions.lua

```diff
@@ -81,9 +87,27 @@ local function lay_down(player, pos, bed_pos, state, skip)
 
 	-- lay down
 	else
+
 		beds.pos[name] = pos
+		meta = minetest.get_meta(beds.pos[name])
+
+		-- Check if bed is occupied
+		for _, other_pos in pairs(beds.bed_position) do
+			if meta:get_string("occupiedby") ~= nil then
```

Comment:
Did you test this?
According to `lua_api` you should rather compare with `""`:
https://github.com/minetest/minetest/blob/0c08f948d7014e66d5a79d7584c03164af2edd93/doc/lua_api.txt#L5628

Reply:
Yeah I did test it, but it's not working because of some other code; I might just revert it to the previous, as that was working fine.

Reply:
Why do you store the name in meta anyways? Can't you just use a table with hashed node positions as keys?

Reply:
Could do; that would work; but the previous system was working so I see no reason to change, really.